stdlib/public/core/StringObject.swift

```diff
@@ -952,9 +965,12 @@ extension _StringObject {
   internal func getSharedUTF8Start() -> UnsafePointer<UInt8> {
     _internalInvariant(largeFastIsShared)
 #if _runtime(_ObjC)
-    if largeIsCocoa {
-      return withCocoaObject {
-        stableCocoaUTF8Pointer($0)._unsafelyUnwrappedUnchecked
+    return withCocoaObject {
+      if largeFastIsConstantCocoa {
+        return _getNSCFConstantStringContentsPointer($0)
```

Comment:
This avoids both the non-inlined call to `stableCocoaUTF8Pointer` and the ObjC msg send inside it to call _fastCStringContents: